evm-utils/evm-state/src/storage/mod.rs

```diff
@@ -627,6 +689,65 @@ impl Storage<OptimisticTransactionDBInner> {
         engine.create_new_backup_flush(self.db.as_ref(), true)?;
         Ok(backup_dir)
     }
+
+    pub fn set_initial(
+        &mut self,
+        accounts: impl IntoIterator<Item = (H256, evm::backend::MemoryAccount)>,
+        state_root: H256,
+    ) -> H256 {
+        use std::collections::hash_map::Entry::*;
+
+        fn set_account_state(
+            state_updates: &mut ChangedState,
+            address: H256,
+            account_state: AccountState,
+        ) {
+            match state_updates.entry(address) {
+                Occupied(mut e) => {
+                    e.get_mut().0 = Maybe::Just(account_state);
+                }
+                Vacant(e) => {
+                    e.insert((Maybe::Just(account_state), HashMap::new()));
+                }
+            };
+        }
+
+        fn ext_storage(
+            state_updates: &mut ChangedState,
+            address: H256,
+            indexed_values: impl IntoIterator<Item = (H256, H256)>,
+        ) {
+            let (_, storage) = state_updates
+                .entry(address)
+                .or_insert_with(|| (Maybe::Just(AccountState::default()), HashMap::new()));
+
+            storage.extend(indexed_values);
+        }
+
+        let mut state_updates: ChangedState = HashMap::new();
+
+        for (
+            address,
+            evm::backend::MemoryAccount {
+                nonce,
+                balance,
+                storage,
+                code,
+            },
+        ) in accounts
+        {
+            let account_state = AccountState {
+                nonce,
+                balance,
+                code: code.into(),
+            };
+
+            set_account_state(&mut state_updates, address, account_state);
+            ext_storage(&mut state_updates, address, storage);
```

Comment:
The same is above, i think fn ext_storage is redundant considering it is created inside a function and only used once
```suggestion
            state_updates
                .entry(address)
                .or_insert_with(|| (Maybe::Just(AccountState::default()), HashMap::new()))
                .1.extend(storage);
```

Reply:
Good call! Fixed at [b30ef6b](https://github.com/velas/velas-chain/pull/417/commits/b30ef6b4df950e1f2aab75f104b86e784f635a42)